indexing-service/src/main/java/org/apache/druid/indexing/overlord/HeapMemoryTaskStorage.java

```diff
@@ -194,6 +194,26 @@ public void setStatus(TaskStatus status)
     }
   }
 
+  @Override
+  public List<Task> getActiveTasksByDatasource(String datasource)
+  {
+    giant.lock();
+
+    try {
+      final ImmutableList.Builder<Task> listBuilder = ImmutableList.builder();
+      tasks.values()
+           .stream()
```

Comment:
Recently we've seen a couple of issues with stream slowness: https://github.com/apache/incubator-druid/pull/8466#discussion_r321338441

The task retrieval code would be called fairly often being part of the supervisor's loop, suggest trying for loops instead of streams here and in MetadataTaskStorage and seeing how that performs

Reply:
It make sense and I will change it.